iosMath/LatexToImage/UIView+Image.m

```diff
@@ -0,0 +1,62 @@
+//
+//  UIView+Image.m
+//  iosMath
+//
+//  Created by wpstarnice on 16/9/19.
+//
+//
+
+#import "UIView+Image.h"
+
+@implementation UIView (Image)
+
+- (UIImage *)generateLatexImage {
+
+    UIGraphicsBeginImageContextWithOptions(self.bounds.size, NO, 0.0);
+    // Make the CALayer to draw in "canvas".
+    [self.layer renderInContext: UIGraphicsGetCurrentContext()];
+    
+    // Fetch an UIImage of our "canvas".
+    UIImage *image = UIGraphicsGetImageFromCurrentImageContext();
+    
+    UIGraphicsEndImageContext();
+    
+    UIImage *flippedImage = [self rotateImage:image onDegrees:0];
```

Comment:
There is no reason to rotate the image.
You can just set:

layer.geometryFlipped = NO;

or better yet just set it on the current context.